core/src/main/java/org/mule/runtime/core/api/streaming/bytes/AbstractCursorStreamProvider.java

```diff
@@ -28,21 +33,45 @@
 
   private final AtomicBoolean closed = new AtomicBoolean(false);
 
+  private final ComponentLocation originatingLocation;
+
+  private Exception closerResponsible;
+
+  private static final boolean TRACK_CURSOR_PROVIDER_CLOSE = getBoolean(TRACK_CURSOR_PROVIDER_CLOSE_PROPERTY);
```

Comment:
the SP should not be consumed here. This should be just a final instance attribute provided by the constructor. Otherwise, we don't only loose the ability to spot enable it, it also requires ugly reflection on the test cases. The SystemProperty should be consumed on the AbstractCursorProviderFactory